boto3/boto3/docs/resource.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from botocore import xform_name
+from botocore.docs.utils import get_official_service_name
+
+from boto3.docs.base import BaseDocumenter
+from boto3.docs.action import ActionDocumenter
+from boto3.docs.waiter import WaiterResourceDocumenter
+from boto3.docs.collection import CollectionDocumenter
+from boto3.docs.subresource import SubResourceDocumenter
+from boto3.docs.attr import document_attribute
+from boto3.docs.attr import document_identifier
+from boto3.docs.attr import document_reference
+from boto3.docs.utils import get_identifier_args_for_signature
+from boto3.docs.utils import get_identifier_values_for_example
+from boto3.docs.utils import get_identifier_description
+from boto3.docs.utils import add_resource_type_overview
+
+
+class ResourceDocumenter(BaseDocumenter):
+    def __init__(self, resource, botocore_session):
+        super(ResourceDocumenter, self).__init__(resource)
+        self._botocore_session = botocore_session
+
+    def document_resource(self, section):
+        self._add_title(section)
+        self._add_intro(section)
+        overview_section = section.add_new_section('member-overview')
+        self._add_identifiers(section)
+        self._add_attributes(section)
+        self._add_references(section)
+        self._add_actions(section)
+        self._add_sub_resources(section)
+        self._add_collections(section)
+        self._add_waiters(section)
+        self._add_overview_of_members(overview_section)
+
+    def _add_title(self, section):
+        section.style.h2(self._resource_name)
+
+    def _add_intro(self, section):
+        identifier_names = []
+        if self._resource_model.identifiers:
+            for identifier in self._resource_model.identifiers:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)